### ISSUE TITLE
Spring Cleaning: Nü Note Options

### ISSUE THREAD

**Issue by jleandroperez** [Automattic/simplenote-ios#763]
### Guidelines
- New code should always be Swift
- Autolayout **is always preferred**
- Please drop outdated components when appropriate! 🔥🔥🔥
- [Simplenote Icons Here](https://github.com/Automattic/Simplecons)

### Details:
- [x] Implement a new Note Options component, and replace the legacy UI

### Mockups
<img width="300" src="https://user-images.githubusercontent.com/1195260/95608947-4a88f980-0a34-11eb-8aa1-e59b538bad9c.png">

**Comment by mattdc28** [comment]
Currently, the note options menu is activated by tapping the `i` button. These mocks show the new menu being presented from a new ellipsis button but the `i` button is still visible. 

Do we want to retain the `i` button for displaying things like word count that aren't displayed in the new menu or should it be removed?

**Comment by jleandroperez** [comment]
(1 + 3) Word Count [is actually another issue](https://github.com/Automattic/simplenote-ios/issues/836). 

For the purpose of this particular task, please, feel free to leave the `( i )` button setup as a placeholder. We'll merge back into *develop* only once all of the pieces are in place.

By the way, every icon you might possibly need [is in this repository](https://github.com/Automattic/Simplecons)! (description also updated with the link).

(2) 100% your call, no preferences there! Whatever approach you think produces the best quality code 😃 

(4) Detail there is: not all of those items will push. Will get back to you on this one!

(5) Since the parent ViewController will have to present the History ([Work In Progress here!](https://github.com/Automattic/simplenote-ios/pull/835)), yes, you'd need to relay a message back.

The approach you take for (2) and (5) will probably be related, if you go for consistency.

Thank you @mattdc28 !!

**Comment by jleandroperez** [comment]
**Addendum:**
I've noticed there was a missing mockup, with an alert (indicating if the Publish OP failed). Such component is (allegedly) similar to this [WordPress iOS component](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/ViewRelated/System/Notices/NoticePresenter.swift), but bringing all of the tooling we've got there might be an overkill.
